docs/use_cases/retrieval_augmented_generation_eval_qdrant_ragas.md

```diff
@@ -0,0 +1,692 @@
+  Use technical terminology appropriately and maintain clarity throughout the response.
+  Based on this Context : {context}
+  """
+        
+  response = client.chat.completions.create(
+        model = model,
+        messages = [ 
+        {'role': 'user', 'content': prompt}
+          ],
+        temperature=0.5,
+        max_tokens=200,
+        n=1,
+        stop=None
+    )
+  return response.choices[0].message.content
+```
+
+We use the subroutine above to generate question and answer (ground-truth) pairs as follows:
+
+```python
+# Generate question-answer pairs for the given chunk
+
+context = context = """HNSW is the approximate nearest neighbor search. 
+          This means our accuracy improves up to a point of diminishing returns, as we check the index for more similar candidates. 
+          In the context of binary quantization, this is referred to as the oversampling rate."""
+
+question_answer_pair = generate_question_answer(context)
+print(question_answer_pair)
+
+## Outputs 
+#Question: How does the oversampling rate relate to the accuracy of approximate nearest neighbor search in the context of binary quantization?
+#Answer: In the context of binary quantization, the oversampling rate refers to the number of similar candidates that are checked in the index during an approximate nearest neighbor search. As we increase the oversampling rate, we are essentially checking more candidates for similarity, which can improve the accuracy of the search up to a certain point. However, there is a point of diminishing returns where further increasing the oversampling rate may not significantly improve the accuracy any further. Therefore, the oversampling rate plays a crucial role in balancing the trade-off between accuracy and computational efficiency in approximate nearest neighbor search using binary quantization.
+```
+
+Using OpenAI to generate your evaluation dataset lets you define the prompt and questions based on the intended use case. Still, like T5, OpenAI requires a nontrivial amount of manual effort to identify and extract correct answers. Let's see how RAGAS compares.
+
+
+**Method 3 -  Using RAGAS**
+
+Compared with T5 and OpenAI, RAGAS is an easier way of generating a Question-Context-Ground_Truth set and a complete baseline evaluation dataset - [utility](https://docs.ragas.io/en/latest/getstarted/testset_generation.html) - that can be created from your data in just a couple of lines of code, as we demonstrate below. 
+In RAGAS, questions with different characteristics such as reasoning, conditioning, multi-context, etc. are systematically crafted from a set of documents. This ensures a more robust evaluation process by providing comprehensive coverage of the performance of various components within your RAG pipeline.
+
+```python
+## Test Evaluation Dataset Generation using Ragas
+from langchain.docstore.document import Document as LangchainDocument
```

Comment:
I think you maybe lost changes you made here when you made your second commit. Should this be:

```python
## Test Evaluation Dataset Generation using Ragas
from ragas.testset.generator import TestsetGenerator
from ragas.testset.evolutions import simple, reasoning, multi_context
from langchain_openai import ChatOpenAI, OpenAIEmbeddings

Reply:
Added back : 
`from langchain_openai import ChatOpenAI, OpenAIEmbeddings`